src/app/page/component/page/cyoa-card-collection-page/cyoa-card-collection-page.component.ts

```diff
@@ -0,0 +1,179 @@
+    this.isForm$ = this.pageService.isForm$;
+    this.isModal$ = this.pageService.isModal$;
+    this.formAction$ = this.pageService.formAction$;
+  }
+
+  ngOnDestroy() {
+    this._unsubscribeAll.next();
+    this._unsubscribeAll.complete();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'page':
+            if (
+              !changes['page'].previousValue ||
+              changes['page'].currentValue !== changes['page'].previousValue
+            ) {
+              this.ready = false;
+              this._page = this.page;
+              this.cards = [];
+              this.init();
+            }
+            break;
+          default:
+            break;
+        }
+      }
+    }
+  }
+  currentCardIndex = 0;
+
+  get currentCard(): CYOAPageCard | undefined {
+    return this.cards?.[this.currentCardIndex];
+  }
+
+  goToCard(index: number): void {
+    if (!this.cards || this.cards.length === 0) return;
+
+    const safeIndex = Math.max(0, Math.min(index, this.cards.length - 1));
+
+    if (safeIndex !== this.currentCardIndex) {
+      this.currentCardIndex = safeIndex;
+      this.updateUrl();
+      this.updateCardState();
+    }
+  }
+
+  showPreviousCard(): void {
+    this.goToCard(this.currentCardIndex - 1);
+  }
+
+  showNextCard(): void {
+    this.goToCard(this.currentCardIndex + 1);
+  }
+
+  private updateUrl(): void {
+    this.router.navigate(['../', this.currentCardIndex], {
+      relativeTo: this.route
+    });
+  }
+
+  private updateCardState(): void {
+    this.isFirstCard = this.currentCardIndex === 0;
+    this.isLastCard = this.currentCardIndex === this.totalCards - 1;
+  }
+
+  private subscribeToCardPosition(): void {
+    this.route.paramMap
+      .pipe(takeUntil(this._unsubscribeAll))
+      .subscribe((params: ParamMap) => {
+        const param = params.get('cardPosition');
+        const index = parseInt(param, 10);
+
+        if (param === null || isNaN(index) || index < 0) {
+          this.router.navigate([0], {
+            relativeTo: this.route,
+            replaceUrl: true
+          });
+        } else {
+          this.goToCard(index);
+        }
+      });
+  }
+
+  private init(): void {
+    console.log('CYOA Card Collection Page:', this._page);
+    console.log('Order:', this.order);
+    console.log('Total Pages:', this.totalPages);
+    this.pageService.setPageOrder(this.order, this.totalPages);
+    this.pageService.modalHidden();
+    this.pageService.formHidden();
+    this.cards = this._page.cards;
+    this.totalCards = this.cards.length || 0;
+    this.showBackButton = !!this._page.parentPage?.position;
+    this.subscribeToCardPosition();
+
+    this.ready = true;
+  }
+
+  navigateBack(): void {
+    if (!this.ready || !this.showBackButton) {
+      return;
+    }
+
+    this.pageService.navigateToPage(this._page.parentPage.position);
+  }
```

Comment:
Same here, we prob should put this in a helper function to help maintain this code